[APM]Update explanation for rare sampler for upcoming changes (#16125)
DataDog/documentation: content/en/tracing/trace_pipeline/ingestion_mechanisms.md

The rare sampler sends a set of rare spans to Datadog. It catches combinations of `env`, `service`, `name`, `resource`, `error.type`, and `http.status` up to 5 traces per second (per Agent). It ensures visibility on low traffic resources when the head-based sampling rate is low.

**Note**: The rare sampler captures local traces at the Agent level. If the trace is distributed, there is no way to guarantee that the complete trace will be sent to Datadog.

#### Datadog Agent 7.41.0 and higher

By default, the rare sampler is **not enabled**. 

> Note: When **enabled**, spans dropped by tracing library rules or custom logic such as `manual.keep` will be evaluated under this sampler.

Author: Thanks for clarifying, is there a `datadog.yaml` version of this configuration?

(I updated the wording a bit, let me know if it looks better now. It will take several minutes for the preview to update though)

The rare sampler sends a set of rare spans to Datadog. It catches combinations of `env`, `service`, `name`, `resource`, `error.type`, and `http.status` up to 5 traces per second (per Agent). It ensures visibility on low traffic resources when the head-based sampling rate is low.

**Note**: The rare sampler captures local traces at the Agent level. If the trace is distributed, there is no way to guarantee that the complete trace will be sent to Datadog.

#### Datadog Agent 6/7.41.0 and higher

By default, the rare sampler is **not enabled**. 

> Note: When **enabled**, spans dropped by tracing library rules or custom logic such as `manual.keep` are **excluded** under this sampler.

Reviewer: ```suggestion
**Note**: When **enabled**, spans dropped by tracing library rules or custom logic such as `manual.keep` are **excluded** under this sampler.
```

To configure the rare sampler, update the `apm_config.enable_rare_sampler` setting in the Agent main configuration file (`datadog.yaml`) or with the environment variable `DD_APM_ENABLE_RARE_SAMPLER`:

```
@params apm_config.enable_rare_sampler - boolean - optional - default: false
@env DD_APM_ENABLE_RARE_SAMPLER - boolean - optional - default: false
```

> To adjust the rare sampler so that it evaluates spans dropped by tracing library rules or custom logic such as `manual.keep`, enable the feature with: `DD_APM_FEATURES=error_rare_sample_tracer_drop` in the Trace Agent.

Reviewer: ```suggestion
To evaluate spans dropped by tracing library rules or custom logic such as `manual.keep`,
 enable the feature with: `DD_APM_FEATURES=error_rare_sample_tracer_drop` in the Trace Agent.
```

#### Datadog Agent 6/7.33 to 6/7.40.x

By default, the rare sampler is enabled.

> Note: When **enabled**, spans dropped by tracing library rules or custom logic such as `manual.keep` **are excluded** under this sampler. To include these spans in this logic, upgrade to Datadog Agent 6.41.0/7.41.0 or higher.

Reviewer: ```suggestion
**Note**: When **enabled**, spans dropped by tracing library rules or custom logic such as `manual.keep` **are excluded** under this sampler. To include these spans in this logic, upgrade to Datadog Agent 6.41.0/7.41.0 or higher.
```
